packages/wallets/src/hooks/useInputATMFormatter.ts

```diff
@@ -66,16 +105,30 @@ const useInputATMFormatter = (initial?: number, options?: TOptions) => {
 
             return onChangeDecimal({ target: { value: unformatted } });
         },
-        [locale, fractionDigits, onChangeDecimal]
+        [input, maxDigits, prevFormattedValue, locale, fractionDigits, onChangeDecimal]
     );
 
     const onPaste: React.ClipboardEventHandler<HTMLInputElement> = useCallback(
-        e => (isPasting.current = e.type === 'paste'),
-        []
+        e => {
+            isPasting.current = e.type === 'paste';
+            if (Number(unFormatLocaleString(formattedValue, locale)) === 0) {
+                const pasted = (e.clipboardData || window.clipboardData).getData('Text');
+                const pastedValue = Number(unFormatLocaleString(pasted, locale));
+                if (!isNaN(pastedValue) && isFinite(pastedValue))
+                    onChange({
+                        target: {
```

Comment:
for future PRs - probably good to avoid objects which pretends to be an events/have Event structure, but actually they are not Events (I know this pattern was already introduced before this PR)